x-pack/plugins/endpoint/public/applications/endpoint/view/alerts/index.tsx

```diff
@@ -82,28 +83,39 @@ export const AlertIndex = memo(() => {
   }, []);
 
   const { pageIndex, pageSize, total } = useAlertListSelector(selectors.alertListPagination);
-  const urlFromNewPageSizeParam = useAlertListSelector(selectors.urlFromNewPageSizeParam);
-  const urlWithSelectedAlert = useAlertListSelector(selectors.urlWithSelectedAlert);
-  const urlWithoutSelectedAlert = useAlertListSelector(selectors.urlWithoutSelectedAlert);
-  const urlFromNewPageIndexParam = useAlertListSelector(selectors.urlFromNewPageIndexParam);
   const alertListData = useAlertListSelector(selectors.alertListData);
   const hasSelectedAlert = useAlertListSelector(selectors.hasSelectedAlert);
+  const queryParams = useAlertListSelector(selectors.uiQueryParams);
 
   const onChangeItemsPerPage = useCallback(
-    newPageSize => history.push(urlFromNewPageSizeParam(newPageSize)),
-    [history, urlFromNewPageSizeParam]
+    newPageSize => {
+      const newQueryParms = { ...queryParams };
+      newQueryParms.page_size = newPageSize;
+      delete newQueryParms.page_index;
+      const relativeURL = urlFromQueryParams(newQueryParms);
+      return history.push(relativeURL);
+    },
+    [history, queryParams]
   );
 
   const onChangePage = useCallback(
-    newPageIndex => history.push(urlFromNewPageIndexParam(newPageIndex)),
-    [history, urlFromNewPageIndexParam]
+    newPageIndex => {
+      return history.push(
+        urlFromQueryParams({
```

Comment:
@paul-tavares doing something closer to what you have.
@dplumlee @peluja1012 thoughts?

Reply:
this works for me. maybe we should also use the same querystring package that @paul-tavares is using (`query-string`) just to keep it consistent. I think it's in kibana already and it's a bit better than the one we're using

Reply:
@oatkiller looks good but it doesn't seem to be working when only `selected_alert` is present. The user gets redirected to `/app/endpoint/`:
![close_flyout_bug mov](https://user-images.githubusercontent.com/616158/75016578-526e3c00-5459-11ea-8365-732df9a3114a.gif)

Reply:
I goofed this up. I'm working on another slight improvement, but I feel like I might just be getting in the way of stuff @paul-tavares is doing. Maybe worth syncing

Reply:
Fixed